langchain/memory/chat_message_histories/redis.py

```diff
@@ -0,0 +1,69 @@
+import json
+import logging
+from typing import List, Optional
+
+from langchain.schema import (
+    AIMessage,
+    BaseChatMessageHistory,
+    BaseMessage,
+    HumanMessage,
+    _message_to_dict,
+    messages_from_dict,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class RedisChatMessageHistory(BaseChatMessageHistory):
+    def __init__(
+        self,
+        session_id: str,
+        url: str = "redis://localhost:6379/0",
+        key_prefix: str = "message_store:",
+        ttl: Optional[int] = None,
+    ):
+        try:
+            import redis
+        except ImportError:
+            raise ValueError(
+                "Could not import redis python package. "
+                "Please install it with `pip install redis`."
+            )
+
+        try:
+            self.redis_client = redis.Redis.from_url(url=url)
+        except redis.exceptions.ConnectionError as error:
+            logger.error(error)
+
+        self.session_id = session_id
+        self.key_prefix = key_prefix
+        self.ttl = ttl
+
+    @property
+    def key(self) -> str:
+        """Construct the record key to use"""
+        return self.key_prefix + self.session_id
+
+    @property
+    def messages(self) -> List[BaseMessage]:  # type: ignore
+        """Retrieve the messages from Redis"""
+        _items = self.redis_client.lrange(self.key, 0, -1)
```

Comment:
@hwchase17, @tylerhutcherson , @3coins,
I know this just got released, but trying it out just found a bug with it.

Unfortunately the use of `self.redis_client.lrange(self.key, 0, -1)` returns the messages in the list in a RANDOM ORDER, not in the order how it was inserted.

I haven't yet looked into how to fix this, just wanted to let everybody know.